mir/src/passes/mod.rs

```diff
@@ -450,3 +473,262 @@ pub fn duplicate_node_or_replace(
         Op::None(_) => {},
     }
 }
+
+/// Helper function to extract the constant felt value from a Link<Op> if it is one.
+pub fn get_inner_const(value: &Link<Op>) -> Option<u64> {
+    match value.borrow().deref() {
+        Op::Value(Value {
+            value:
+                SpannedMirValue {
+                    value: MirValue::Constant(ConstantValue::Felt(c)),
+                    ..
+                },
+            ..
+        }) => Some(*c),
+        _ => None,
+    }
+}
+
+/// Handle the visit of an accessor node, used for both Unrolling and ConstantPropagation passes
+/// The `compute_indices` bool indicates whether we the indices need to be known constant at this
+/// stage.
+pub fn handle_accessor_visit(
+    accessor: Link<Op>,
+    compute_indices: bool,
```

Comment:
nit (feel free to ignore): I would maybe invert `compute_indices` into something like `expect_constant_indices`. May make the meaning a bit more clear (at least that's how we describe it in the doc comments).